server/plugin.go

```diff
@@ -314,6 +316,55 @@ func (p *Plugin) getZoomUserWithToken(token *oauth2.Token) (*zoom.User, error) {
 	return &zoomUser, nil
 }
 
+func (p *Plugin) GetMeetingOAuth(meetingID int, userID string) (*zoom.Meeting, error) {
+	config := p.getConfiguration()
+	ctx := context.Background()
+
+	conf, err := p.getOAuthConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	zoomUserInfo, apiErr := p.getZoomUserInfo(userID)
+	if apiErr != nil {
+		return nil, apiErr
+	}
+
+	client := conf.Client(ctx, zoomUserInfo.OAuthToken)
+	apiURL := config.ZoomAPIURL
+	if apiURL == "" {
+		apiURL = zoomDefaultAPIURL
+	}
+
+	url := fmt.Sprintf("%v/meetings/%v", apiURL, meetingID)
+	res, err := client.Get(url)
+	if err != nil {
+		return nil, errors.New("error fetching zoom user, err=" + err.Error())
+	}
+	if res == nil {
+		return nil, errors.New("error fetching zoom user, empty result returned")
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("error fetching zoom user")
+	}
+
+	buf := new(bytes.Buffer)
+
+	if _, err = buf.ReadFrom(res.Body); err != nil {
+		return nil, errors.New("error reading response body for zoom user")
+	}
```

Comment:
Nit: Using `ioutil.ReadAll` is the more common way to read a `Body` given that it's then unnecessary to allocate it.